wandb/apis/public/api.py

```diff
@@ -1048,3 +1048,24 @@ def list_jobs(self, entity, project):
             return [x["node"]["artifacts"] for x in artifacts]
         except requests.exceptions.HTTPError:
             return False
+
+    @normalize_exceptions
+    def exists(self, name, type=None):
+        _, _, artifact_name = self._parse_artifact_path(name)
+        if ":" in artifact_name:
+            try:
+                self.artifact(name, type)
+                return True
+            except wandb.errors.CommError:
+                return False
+        else:
+            if type is None:
```

Comment:
For this PR I think it's fine, but I don't know why we require a type to search for artifacts. It shouldn't be an issue to update the underlying functions / queries to allow `type=None`.